pybossa/view/projects.py

```diff
@@ -1551,6 +1552,22 @@ def respond():
                 for col in disp_info_columns:
                     task['info'][col] = task_info.get(col, '')
 
+        if 'lock_status' in args['display_columns']:
+            user_info = {}
+            for task in page_tasks:
+                users = []
+                for user_id in task['lock_users']:
+                    try:
+                        # show fullname for users
+                        user_id = int(user_id)
+                        if not user_info.get(user_id):
+                            user_info[user_id] = cached_users.get_user_by_id(user_id).fullname
+                        users.append(user_info[user_id])
+                    except ValueError:
+                        # show ip address for anonymous users
+                        users.append(user_id)
```

Comment:
we don't support anonymous user access. Should this be raising exception instead?

Reply:
Raising an exception would cause the task browse page to crush, are there more graceful ways to handle the situation?

On the other hand we have validations to prevent anonymous workers from locking tasks so we can pretty much ensure all locked users are logged in.